### ISSUE TITLE
Vendor packages are not resolved by IntelliJ idea when using gogradle.

### ISSUE THREAD

**Issue by darkpssngr** [gogradle/gogradle#152]
I'm not sure if this is an issue with the GoGradle plugin or IntelliJ IDEA. Even though gradle build works, IDEA is not able to resolve packages and so does not give code completion.

I suspect this might be a bug in IDEA with using custom project GOPATH which gogradle sets. 
Is someone facing a similar issue/ can anyone help with this?

Thanks

**Comment by blindpirate** [comment]
Yes, you're right, there's some issues about project-level GOPATH: https://youtrack.jetbrains.com/oauth?state=%2Fissue%2FGO-4118

Could you try global GOPATH and make your project structure conform to [go project layout](https://golang.org/cmd/go/#hdr-GOPATH_environment_variable) and try again?

**Comment by blindpirate** [comment]
And looks like IDEA 2017.2 no longer supports go plugin...https://github.com/go-lang-plugin-org/go-lang-idea-plugin/issues/2898

**Comment by darkpssngr** [comment]
Sorry, forgot to mention I'm using the Ultimate version. 
I tried setting it to the global GOPATH, still does not work.

**Comment by zhanghuabin** [comment]
After some investigations, I found the root cause of this issue:

## The `vendor` folder created by gogradle does not contain a `src` sub-folder.

Please see some screenshots as below:

![ss1](https://user-images.githubusercontent.com/5813426/36832648-51c3f496-1d67-11e8-9aef-bdf403bb3d5a.png)
![ss2](https://user-images.githubusercontent.com/5813426/36832733-aa882dc2-1d67-11e8-95c9-fe9bff1a6e3d.png)

The IDEA golang plugin will not recognize libs in `vendor` folder with above project layout and IDE settings.

But if I add a sub-folder `src` in `vendor`:
![ss4](https://user-images.githubusercontent.com/5813426/36833141-5910f0c6-1d69-11e8-8a0e-57adad0a39d5.png)

Golang plugin works well.

I guess that the golang plugin consider that all paths in GOPATH **MUST** have a `src` sub-folder by default.

So you may add a new system property to change the path of `vendor`, which could be relative or absolute.

Thanks a lot
Huabin Zhang

**Comment by fiftoine** [comment]
@zhanghuabin what do you mean by : 

> So you may add a new system property to change the path of vendor, which could be relative or absolute.

Could you provide an example on how to achieve this? 
Thanks in advance

**Comment by kiview** [comment]
I'm having the same problem using the Go plugin in IDEA 2018.2.
Any workaround to solve this? Build with gradle is working fine.

**Comment by kiview** [comment]
@blindpirate Thanks for the suggestion. I have actually the same problem with GoLand.
Here is my config, as describe in the docs:
![goland_gopath](https://user-images.githubusercontent.com/5088104/47014047-5ee9d280-d149-11e8-8f4a-406db2e67fb7.png)
![goland_goroot](https://user-images.githubusercontent.com/5088104/47014048-5ee9d280-d149-11e8-8c35-1b5ccb603544.png)
![goland_project](https://user-images.githubusercontent.com/5088104/47014049-5ee9d280-d149-11e8-9398-c6e140438b89.png)

I also tried adding the vendor directly to the GOPATH directly, as well as the project root, without success.

**Comment by blindpirate** [comment]
> My suggestion would be GoLand and global GOPATH.

Note the **global** GOPATH.

@kiview I assume you're familiar with `GOPATH`. The way Gogradle implements project-level `GOPATH` is symlink, which in most cases works well - except for GoLand. I investigated several issues before (see [here](https://youtrack.jetbrains.com/issue/GO-4118)).

**Comment by kiview** [comment]
Maybe I misunderstand your remark, but doesn't the top screenshot show the config for the global GOPATH?

**Comment by blindpirate** [comment]
It's not. The GOPATH in the first screenshot exists in the project directory - it's a "project-level"  GOPATH - i.e. it's only used by that project. If you take a look at the directory you can find a symlik in `<your project>/.gogradle/project_gopath/src/<your package path>`. The name `project_gopath` indicates it's not global.

By global, I mean [this doc](https://golang.org/doc/code.html#GOPATH):

> The GOPATH environment variable specifies the location of your workspace.

**Comment by kiview** [comment]
@blindpirate Thanks a lot for your help. By studying the plugin's source code and the Go docs I was able to figure it out (I don't have that much Go experience).

It works fine with Go Plugin from IntelliJ ultimate by the way (which should be based on the same code base as GoLand).

**Comment by deepu105** [comment]
I can confirm similar issue when using VSCode with a gogradle project as well. If I move all content of vendor into a folder named `src` then everything works else VSCode doesn't see any of the vendor libs. Can go gradle provide a way to use an src folder?

**Comment by MurtadhaS** [comment]
For any one facing this issue, just tick the "Index entire GOPATH" option in Preferences->Go->GOPATH
